### PR TITLE
Add support for config file in Go Automation api

### PR DESCRIPTION
**Pull request by zbuchheit**
Related: https://github.com/pulumi/pulumi/issues/17773

This adds support for the config file flag in the Go Automation API. It adds the option to preview, up, refresh, and destroy and also adds the ability to edit/get a config file via the workspace.

I had to edit some local_workspace_test tests to account for a mock change. Previously capturedArgs would only be set to the last set of args. This is a problem to use with up as the last thing to happen is pulumi output to be called. I changed this type to [][]string to allow us to capture all calls made with the mock rather than the last call.

### CODE REVIEW

File: sdk/go/auto/workspace.go [hunk 1]
```diff
@@ -61,6 +61,8 @@ type Workspace interface {
 	GetConfigWithOptions(context.Context, string, string, *ConfigOptions) (ConfigValue, error)
 	// GetAllConfig returns the config map for the specified stack name, scoped to the current workspace.
 	GetAllConfig(context.Context, string) (ConfigMap, error)
+	// GetAllConfig returns the config map for the specified stack name, scoped to the current workspace.
```
Comment (Frassle) [inline sdk/go/auto/workspace.go:64]: Doc comment needs updating

File: sdk/go/auto/local_workspace.go [hunk 2]
```diff
@@ -232,6 +235,22 @@ func (l *LocalWorkspace) GetAllConfig(ctx context.Context, stackName string) (Co
 	return val, nil
 }
 
+// GetAllConfigFromFile returns the config map for the specified stack name, scoped to the current workspace.
+// LocalWorkspace reads this config from the matching file.
+func (l *LocalWorkspace) GetAllConfigFromFile(ctx context.Context, stackName string, path string) (ConfigMap, error) {
```
[the remark this answers is not in the document]
Comment (zbuchheit) [reply sdk/go/auto/local_workspace.go]: I reverted those changes, good call out

File: sdk/go/auto/local_workspace.go [hunk 3]
```diff
@@ -232,6 +235,22 @@ func (l *LocalWorkspace) GetAllConfig(ctx context.Context, stackName string) (Co
...
+	var val ConfigMap
+	stdout, stderr, errCode, err := l.runPulumiCmdSync(ctx,
+		"config", "--show-secrets", "--json", "--stack", stackName, "--config-file", path)
```
[the remark this answers is not in the document]
Comment (zbuchheit) [reply sdk/go/auto/local_workspace.go]: Change reverted, this should no longer be relevant

File: sdk/go/auto/workspace.go (hunk 1 above)
Comment (zbuchheit) [reply sdk/go/auto/workspace.go]: I ended up removing the Get and Set config functionality from this PR as it probably should be tracked separately

**Comment by zbuchheit** [comment]
> We should probably implement this in Python and NodeJS at the same time (or at least open separate issues for those, so we don't lose track of disparities)

I adjusted https://github.com/pulumi/pulumi/issues/17773 to make sure and track all languages. I am happy to split it up into multiple issues if needed

**Comment by zbuchheit** [comment]
I had to adjust the stack_test.go. I thought the tests would work as they were written, but Up and the other operations expect some stdout data that needs to unmarshal to the summaries, so they were valid in failing. Also worth noting there isn't an equivalent refreshOptsToCmd function for Up or Preview so I wasn't able to easily write tests for those.

File: sdk/go/auto/local_workspace_test.go [hunk 4]
```diff
@@ -66,7 +66,7 @@ type mockPulumiCommand struct {
 	stderr       string
 	exitCode     int
 	err          error
-	capturedArgs []string
+	capturedArgs [][]string
```
Comment (tgummerer) [inline sdk/go/auto/local_workspace_test.go:69]: Not sure I completely understand why this change was necessary.  Did these tests not work before?  How did they change?

**Review by tgummerer — APPROVED** [review]

File: sdk/go/auto/local_workspace_test.go (hunk 4 above)
Comment (zbuchheit) [reply sdk/go/auto/local_workspace_test.go:69]: I was originally changing them to use with Preview, Up, etc, because there isn't an easy way to capture those args as they call runPulumiCmdSync multiple times, so hard to capture and verify those args, but to your point they are no longer relevant for this PR. I'll revert the changes.
